src/lib/parsers/sol-json.c

```diff
@@ -536,9 +528,37 @@ sol_json_double_to_str(const double value, char *buf, size_t len)
         char *fraction = decimal_point + decimal_len;
         *decimal_point = '.';
 
-        memmove(decimal_point + 1, fraction, (buf + ret + 1) - fraction);
+        memmove(decimal_point + 1, fraction, (buf + len + 1) - fraction);
     }
 #endif
+}
+
+SOL_API int
+sol_json_double_to_str(const double value, char *buf, size_t len)
+{
+    int ret;
+
+    SOL_NULL_CHECK(buf, -EINVAL);
+
+    ret = snprintf(buf, len, "%f", value);
```

Comment:
I found a problem with it. 

printf("%g", DBL_MIN); //2.22507e-308

if you try to use the returned string with strtod it will fail due an underflow :(
Well, i'm going to think how to make it works.